docs/develop/plugins/resource_provider_plugins.md

```diff
@@ -176,3 +176,332 @@ module.exports = function (app) {
   }
 }
 ```
+
+---
+
+## Delta Notifications for Resource Changes
+
+Resource provider plugins should emit delta notifications when resources are modified through custom endpoints to keep clients synchronized in real-time. While the built-in Resources API automatically emits deltas for standard operations (`POST`, `PUT`, `DELETE`), custom provider endpoints must manually emit deltas.
+
+### Why Emit Deltas?
+
+Without delta notifications:
+
+- Clients must poll the REST API repeatedly to detect changes
+- Users must manually refresh applications after modifying resources
+- Server restarts may be required for changes to become visible
+- Mobile clients waste battery on unnecessary polling
+
+With delta notifications:
+
+- Changes appear instantly across all connected clients
+- No polling required - updates pushed via WebSocket
+- Better user experience with real-time synchronization
+- Lower bandwidth and battery consumption
+
+### When to Emit Deltas
+
+Emit delta notifications after:
+
+1. **Upload** - New resource added
+2. **Delete** - Resource removed
+3. **Modify** - Resource updated (rename, move, etc.)
+4. **Toggle** - Resource enabled/disabled
```

Comment:
Not sure I understand "toggle" in this in the client context.
What delta message would this action emit? I would expect to the client it would just appear as resource updated or deleted deltas.